homeassistant/components/google/quality_scale.yaml

```diff
@@ -0,0 +1,76 @@
+rules:
+  # Bronze
+  config-flow:
+    status: done
+    comment: Integration uses an OAuth config flow.
+  brands: done
+  dependency-transparency: todo
+  common-modules: done
+  has-entity-name: done
+  action-setup:
+    status: todo
+    comment: |
+      Actions are current setup in `async_setup_entry` and need to be moved
+      to `async_setup`.
+  appropriate-polling: done
+  test-before-configure: done
+  entity-event-setup:
+    status: exempt
+    comment: Integration does not subscribe to events.
+  unique-config-entry: done
+  entity-unique-id: done
+  docs-installation-instructions: done
+  docs-removal-instructions: todo
+  test-before-setup:
+    status: todo
+    comment: |
+      The integration does not test the connection in `async_setup` but
+      instead does this in the calendar platform only, which can be improved.
+  docs-high-level-description: done
+  config-flow-test-coverage: done
```

Comment:
I see that we have a test that tests multiple config entries for the device auth method, but shouldn't we also test the application_credentials?

Reply:
My impression is that tests have been added when support for it was added #103570

Reply:
But where is that test then?

Reply:
You're right its only testing reauth not multiple config entries. I consider these orthogonal parts of the config flow and not strictly necessary to test the combination of them.

It definitely sounds like it can't hurt, but not something i'm planning on adding immediately, but can consider if doing additional test improvements in the future.

I consider this meeting the coverage rules both in spirit and following what is written -- let me know if you disagree and would like me to mark this as not completed.

Reply:
I'm learning a new worth today. Nice.

> You're right its only testing reauth not multiple config entries. I consider these orthogonal parts of the config flow and not strictly necessary to test the combination of them.

If I understand you correctly, you mean a test here where we setup one with the device auth and application_credentials? I am not necessarily hinting towards that kind of test, it's more that we have a test for the device auth credentials that hits the `self._abort_if_unique_id_configured()`, but we don't have that one for application credentials (where we setup config entry with same characteristics and then try to setup again to see it abort).

Is that the same thing you were thinking about or not?

Reply:
> If I understand you correctly, you mean a test here where we setup one with the device auth and application_credentials? I am not necessarily hinting towards that kind of test, it's more that we have a test for the device auth credentials that hits the `self._abort_if_unique_id_configured()`, but we don't have that one for application credentials (where we setup config entry with same characteristics and then try to setup again to see it abort).
> 
> Is that the same thing you were thinking about or not?

I believe understand what you're saying and i'm saying I think we only need one test exercising this and the authentication method does not matter. I think it is important to have tests of multiple combinations of factors when the factors are important to test together. Here, I don't think those are meaningful to test together.

(I may do it to end the discussion, but not because i think it is meaningful.)